### ISSUE TITLE
split formatting i18n and i18n from main package

### ISSUE THREAD

**Issue by pke** [opening-hours/opening_hours.js#150]
The lib comes in (minimized) at 115kb at the moment. Most of that is i18n text. Would it be possible to separate that from the core lib so we can `require` it on demand in projects?

**Comment by ypid** [comment]
Sounds good. Should be possible and I am open for PRs.

**Comment by pke** [comment]
Also as a first improvement we could make the "moment" package a `peerDepencency` and not a `dependency`. This would reduce this packages size when its compiled in a bundle like with `webpack` or `browserify`

**Comment by ypid** [comment]
@pke Sounds good. Thanks for the hint. I did not know `peerDepencency` before.

Does https://github.com/opening-hours/opening_hours.js/commit/1c2909977cceadceee37283e39a2217c42c146b0 look good to you?

**Comment by pke** [comment]
It should do the trick. I will try the branch later this week.

**Comment by pke** [comment]
Did you update and publish a new version to npm? Current 3.4.0 does not contain the `peerDependencies`

**Comment by ypid** [comment]
I did not. If you want to help with that, check the [v3.5.0 milestone](https://github.com/opening-hours/opening_hours.js/milestone/10) :wink:

**Comment by pke** [comment]
Ah I see. Well that is way too much unsolved in there and I need a small bundle now. So I will go with my own for for now.
Good luck translating the lib to `haxe`, though :)

**Comment by ypid** [comment]
Thanks :wink:

**Comment by ypid** [comment]
The i18n text has been moved to separate files in #198. How the lib reacts if they are not present is not yet documented I think.

**Comment by ypid** [comment]
@pke Can you check if #198 solves this issue for you? All i18n are still included by default and you would need to come up with some way to make them optional or so.

**Comment by pke** [comment]
I can try to strip out the locales with webpack. Whats the current folder structure? Where are the locales saved?

**Comment by pke** [comment]
No, seems like the main opening_hours.js still contains all the locales. There is no way to strip them out that way.

**Comment by ypid** [comment]
> No, seems like the main opening_hours.js still contains all the locales. There is no way to strip them out that way.

You will need to build your own `opening_hours.js`. The repo does not ship with `opening_hours.js` anymore. (But NPM packages will).

**Comment by pke** [comment]
I'm using the npm version. Can't build it myself

**Comment by ypid** [comment]
The idea is that the NPM package ships with both the universal `opening_hours.js` and ES6 files/modules which can be included directly.

**Comment by pke** [comment]
But it doesn't. or at least I don't see it.

**Comment by ypid** [comment]
True, it has not be released yet and it is not yet clear to me if there is anything to watch out for when including ES6 files alongside and how to include them then.

**Comment by pke** [comment]
I am no longer using this lib, so I am cleaning up my open issues.

**Comment by ypid** [comment]
@pke Just because you don’t use this lib anymore does not mean that you have to close valid issues you reported ;-) Reopening.

For reference, you probably switched to https://github.com/pke/simple-opening-hours. I guess such info is always interested for others reading this.
